docs/supported-sources/google-ads.md

```diff
@@ -0,0 +1,137 @@
+
+You can run the following to achieve this:
+```sh
+ingestr ingest \
+  --source-uri "googleads://12345678?credentials_path=./svc_account.json&dev_token=dev-token-spec-1" \
+  --source-table "campaign_report_daily" \
+  --dest-uri "duckdb://./adverts.db" \
+  --dest-table "public.campaigns"
+```
+
+## Custom Reports
+`googleads` source supports custom reports. You can pass a custom report definition to `--source-table` and it will dynamically create a report for you. These reports are aggregated at a daily interval.
+
+The format of a custom report looks like the following:
+```
+daily:{resource_name}:{dimensions}:{metrics}
+```
+Where:
+* `{resource_name}` is a [Google Ads Resource](https://developers.google.com/google-ads/api/fields/v18/overview_query_builder#list-of-all-resources).
+* `{dimensions}` is a comma separated list of the Resource's attribute fields, or fields of [attributed resources](https://developers.google.com/google-ads/api/docs/query/overview).
+* `{metrics}` is a comma separated list of the Resource's [metrics](https://developers.google.com/google-ads/api/fields/v18/metrics). Note that the `metrics.` prefix is optional. 
+
+Notes:
+* `{dimensions}` and `{metrics}` are optional. If you don't need them, you can leave their respective segment blank.
+* `segments` are currently not supported as dimensions.
+* `segments.date` is automatically added to all custom reports.
+
+### Custom Report Example
+For this example, we will ingest data from `ad_group_ad_asset_view`.
+We want to obtain the following info:
+**dimensions**
+  * ad_group.id
+  * campagin.id
+  * customer.id
+**metrics**
+  * metrics.clicks
+  * metrics.conversions
+  * metrics.impressions
+
+To achieve this, we pass a `custom` report specification to `ingestr` source table as follows:
+```sh
+ingestr ingest \
+  --source-uri "googleads://12345678?credentials_path=./svc_account.json&dev_token=dev-token-spec-1" \
+  --source-table "daily:ad_group_ad_asset_view:ad_group.id,campaign.id,customer.id:clicks,conversions,impressions" \
+  --dest-uri "duckdb:///custom.db" \
+  --dest-table "public.report"
+```
+
+Notice the lack of `metrics.` prefix in the metrics segment. Please note that `--dest-table` is mandatory when creating
+a custom report.
+
+**Without Metrics**
+
+Here's an example of the above report, without any associated metrics:
+```sh
+ingestr ingest \
+  --source-uri "googleads://12345678?credentials_path=./svc_account.json&dev_token=dev-token-spec-1" \
+  --source-table "daily:ad_group_ad_asset_view:ad_group.id,campaign.id,customer.id:" \
+  --dest-uri "duckdb:///custom.db" \
+```
+
+**Without Dimensions**
+
+Here's an example of the above report, without any associated dimensions:
+```sh
+ingestr ingest \
+  --source-uri "googleads://12345678?credentials_path=./svc_account.json&dev_token=dev-token-spec-1" \
+  --source-table "daily:ad_group_ad_asset_view::clicks,conversions,impressions" \
+  --dest-uri "duckdb:///custom.db" \
+```
+
+
+
+## Tables
```

Comment:
maybe this should go above the custom daily reports?

Reply:
done